update.go

```diff
@@ -338,8 +339,22 @@ func (u *Update) FromStream(updateWith io.Reader) (err error, errRecover error)
 			return
 		}
 	case PATCHTYPE_NONE:
-		// no patch to apply, go on through (with bzip2 decoding).
-		newBytes, err = ioutil.ReadAll(bzip2.NewReader(updateWith))
+		// no patch to apply, go on through
+		var fileHeader []byte
+		bufBytes := bufio.NewReader(updateWith)
+		fileHeader, err = bufBytes.Peek(2)
+		if err != nil {
+			return
+		}
+
+		if bytes.Equal([]byte{0x42, 0x5a}, fileHeader) {
```

Comment:
We only use bzip2 files to distribute updates, the fallback case is only used when running tests (because we don't have a way to to bzip2 compression with the stdlib). So we definitely expect a file that begins with the bzip2 magic bytes.